src/csync/csync_reconcile.cpp

```diff
@@ -310,7 +345,13 @@ static int _csync_merge_algorithm_visitor(csync_file_stat_t *cur, CSYNC * ctx) {
                 break;
             case CSYNC_INSTRUCTION_IGNORE:
                 cur->instruction = CSYNC_INSTRUCTION_IGNORE;
-            break;
+                break;
+            case CSYNC_INSTRUCTION_EVAL_RENAME:
+                // Local modification and remote rename:
+                // Typically a parent directory was renamed!
+                // The locally touched file should be uploaded.
+                cur->instruction = CSYNC_INSTRUCTION_NEW;
```

Comment:
Are you sure about the comment?
Here is how i think it can happen:

on the client:
```
echo "foo" > foo  # new file
```
on the server:
```
mv bar foo
```

I suppose this should be a conflict.
Currently this should be handled as such when EVAL_RENAME is handled.

Reply:
I've changed things a bit. The problem is this case:
```
// local
echo "foo" >> A/f
// remote
mv A B
```
then A/f was left with EVAL since B/f is EVAL_RENAME. When we went to the remote tree, there's no other since B/f doesn't exist in local. Then the rename doesn't work and the touched A/f isn't uploaded.

I've added your test case and that just works. The reconcile doesn't care if the instruction is NEW or EVAL.